lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -1108,10 +1109,19 @@ private function shouldProxiedMethodReturn(ReflectionMethod $method)
      * @return string
      */
     private function formatType(
-        ReflectionNamedType $type,
+        ReflectionType $type,
         ReflectionMethod $method,
         ?ReflectionParameter $parameter = null
     ) {
+        if ($type instanceof ReflectionUnionType) {
+            return implode('|', array_map(
+                function (ReflectionType $unionedType) use ($method, $parameter) {
+                    return $this->formatType($unionedType, $method, $parameter);
+                },
+                $type->getTypes()
+            ));
+        }
+
         $name      = $type->getName();
```

Comment:
Maybe add an `assert` before? If it's not a `ReflectionUnionType`, then it must be a `ReflectionNamedType`, right? (although phpstan can't know that for sure).

Another solution might be to use `@param ReflectionUnionType|ReflectionNamedType $type`, but it might cause other issues, I'm not sure.